src/main/java/seedu/task/logic/commands/RedoCommand.java

```diff
@@ -0,0 +1,53 @@
+//@@kellyli A0164103W
+package seedu.task.logic.commands;
+
+import java.util.Optional;
+
+import seedu.task.logic.commands.exceptions.CommandException;
+import seedu.task.logic.history.TaskMemento;
+import seedu.task.model.task.Task;
+import seedu.task.model.task.TaskId;
+import seedu.task.model.task.UniqueTaskList.DuplicateTaskException;
+import seedu.task.model.task.UniqueTaskList.TaskNotFoundException;
+
+public class RedoCommand extends Command {
+    public static final String COMMAND_WORD = "redo";
+
+    public static final String MESSAGE_SUCCESS = "Command redone";
+    public static final String MESSAGE_NO_HISTORY = "No commands to redo";
+
+    @Override
+    public CommandResult execute() throws CommandException {
```

Comment:
Oh, I meant that both tasks within the memento are not null. Sorry, don't know why I put commands